TFNetworkRecLayer.py

```diff
@@ -4383,3 +4383,273 @@ def __call__(self, inputs, state, scope=None):
+      self._fwd_weights_initializer = get_initializer(
+        forward_weights_init, seed=self.network.random.randint(2**31), eval_local_ns={"layer": self})
+    if recurrent_weights_init is not None:
+      self._rec_weights_initializer = get_initializer(
+        recurrent_weights_init, seed=self.network.random.randint(2**31), eval_local_ns={"layer": self})
+    if bias_init is not None:
+      self._bias_initializer = get_initializer(
+        bias_init, seed=self.network.random.randint(2**31), eval_local_ns={"layer": self})
+    if self._rec_weights_initializer:
+      default_var_initializer = self._rec_weights_initializer
+    elif self._fwd_weights_initializer:
+      default_var_initializer = self._fwd_weights_initializer
+    else:
+      default_var_initializer = xavier_initializer(seed=self.network.random.randint(2**31))
+    with reuse_name_scope("rec-twod", initializer=default_var_initializer) as scope:
+      assert isinstance(scope, tf.VariableScope)
+      self._rec_scope = scope
+      scope_name_prefix = scope.name + "/"  # e.g. "layer1/rec/"
+      with self.var_creation_scope():
+        self.cell = self._get_cell(unit_opts=unit_opts)
+
+      # this must not be part of var_creation_scope - otherwise the used operations appear to TF to be used outside
+      # of the while loop, leading to errors
+      y = self._get_output_native_rec_op(self.cell)
+
+      self.output.placeholder = y
+      if self.network.have_rec_step_info():
+        self.output.shape = (self.output.shape[-1],)
+        self.output.time_dim_axis = None
+        self.output.batch_dim_axis = 0
+      else:
+        self.output.time_dim_axis = 0
+        self.output.batch_dim_axis = 1
+
+      # Very generic way to collect all created params.
+      # Note that for the TF RNN cells, there is no other way to do this.
+      # Also, see the usage of :func:`LayerBase.cls_layer_scope`, e.g. for initial vars.
+      params = tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES, scope=re.escape(scope_name_prefix))
+      self._add_params(params=params, scope_name_prefix=scope_name_prefix)
+
+  @classmethod
+  def _post_init_output(cls, output, network, target=None, size_target=None, sources=(), **kwargs):
+    """
+    :param Data output:
+    :param TFNetwork.TFNetwork network:
+    :param str|None target:
+    :param str|None size_target:
+    :param list[LayerBase] sources:
+    """
+    if network.train_flag is not False:
+      # TODO: In training, this is ok. Maybe as well as for eval but not clear.
+      # In forward, mark_data_key_as_used=False should be used and anyway that target value is not available.
+      output.size_placeholder = cls._static_get_target_value(
+        target=target or size_target, network=network,
+        mark_data_key_as_used=network.train_flag is not False).size_placeholder.copy()
+    else:
+      with tf.control_dependencies(None):
+        output.size_placeholder = {} # Not sure why this has to be defined. Is it ever used?
```

Comment:
Follow Python code style conventions, ie. two spaces before `#`.
Also, remove the comment anyway here. The `output.size_placeholder` is used in many places. (Let's discuss offline if you wonder where...)